rosplan_knowledge_base/src/VALVisitorOperator.cpp

```diff
@@ -128,6 +128,7 @@ namespace KCL_rosplan {
 
 	void VALVisitorOperator::visit_effect_lists(VAL::effect_lists * e) {
 
+		eff_time = VAL::E_AT_END;  // Set non-timed add and del effects to be at end in the ROSPlan msgs
```

Comment:
I've been looking at this change today, and this is perfect. If we have the same thing for the condition, but "at start", then instant actions are covered. If possible, can you set the default time for *conditions* to be *at start*?

Reply:
Since conditions are parsed before effects for each action, this could be done at the reset of the visitor.